src/test/groovy/hudson/plugins/gradle/injection/BuildScanInjectionGradleIntegrationTest.groovy

```diff
@@ -443,6 +444,96 @@ class BuildScanInjectionGradleIntegrationTest extends BaseGradleIntegrationTest
         StringUtils.countMatches(JenkinsRule.getLog(secondRun), "ERROR: Gradle Enterprise access key format is not valid") == 1
     }
 
+    def 'vcs repository pattern injection for freestyle remote project - #pattern'(String pattern) {
+        given:
+        def gradleVersion = '8.0.2'
+        gradleInstallationRule.gradleVersion = '8.0.2'
+        gradleInstallationRule.addInstallation()
+
+        DumbSlave slave = createSlave()
+
+        FreeStyleProject p = j.createFreeStyleProject()
+        p.setScm(new GitSCM("https://github.com/c00ler/simple-gradle-project"))
+        p.setAssignedNode(slave)
+
+        p.buildersList.add(new Gradle(tasks: 'clean', gradleName: gradleVersion, switches: "--no-daemon"))
+
+        when:
+        // first build to download Gradle
+        def build = j.buildAndAssertSuccess(p)
+
+        then:
+        j.assertLogNotContains(MSG_INIT_SCRIPT_APPLIED, build)
+
+        when:
+        withInjectionConfig {
+            injectionVcsRepositoryPatterns = pattern
+        }
+        enableBuildInjection(slave, gradleVersion)
+        def build2 = j.buildAndAssertSuccess(p)
+
+        then:
+        if (pattern == "simple-") {
+            j.assertLogContains(MSG_INIT_SCRIPT_APPLIED, build2)
+        } else {
+            j.assertLogNotContains(MSG_INIT_SCRIPT_APPLIED, build2)
+        }
+
+        where:
+        pattern << ["simple-", "this-one-does-not-match"]
```

Comment:
:thought_balloon: could be replaced by a closure param
```
        then:
        pattern == result(j, build2)

        where:
        pattern                   | result
        "simple-"                 | { j, b -> j.assertLogContains(MSG_INIT_SCRIPT_APPLIED, b) }
        "this-one-does-not-match" | { j, b -> j.assertLogNotContains(MSG_INIT_SCRIPT_APPLIED, b) }
```

Reply:
As discussed, wouldn't work :(